### ISSUE TITLE
Java 11 issue

### ISSUE THREAD

**Issue by brettwooldridge** [Netflix/hollow#346]
I have an existing persisted state that my server is trying to restore on startup.  No code has changed, but I have updated to Java 11 from Java 8.  When restoring the state, I am receiving the following exception:
```
 java.lang.RuntimeException: java.util.concurrent.ExecutionException: com.netflix.hollow.api.error.IncompatibleSchemaException: No common schema exists for String: field value has unmatched types: bytes vs string
	at com.netflix.hollow.core.write.HollowWriteStateEngine.restoreFrom(HollowWriteStateEngine.java:169) ~[hollow-2.16.6.jar:2.16.6]
	at com.netflix.hollow.api.producer.HollowProducer$2.restore(HollowProducer.java:283) ~[hollow-2.16.6.jar:2.16.6]
	at com.netflix.hollow.api.producer.HollowProducer.restore(HollowProducer.java:322) ~[hollow-2.16.6.jar:2.16.6]
	at com.netflix.hollow.api.producer.HollowProducer.restore(HollowProducer.java:280) ~[hollow-2.16.6.jar:2.16.6]
	at dancer.service.hollow.internal.HollowEngine.<init>(HollowEngine.kt:126) ~[com-dancer-service-hollow.jar:?]
	at dancer.service.hollow.internal.HollowService$getHollowEngine$1.apply(HollowService.kt:160) ~[com-dancer-service-hollow.jar:?]
	at dancer.service.hollow.internal.HollowService$getHollowEngine$1.apply(HollowService.kt:36) ~[com-dancer-service-hollow.jar:?]
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705) ~[?:?]
	at dancer.service.hollow.internal.HollowService.getHollowEngine(HollowService.kt:160) ~[com-dancer-service-hollow.jar:?]
	at dancer.service.hollow.internal.HollowService.access$getHollowEngine(HollowService.kt:36) ~[com-dancer-service-hollow.jar:?]
	at dancer.service.hollow.internal.HollowService$rebuildState$1.run(HollowService.kt:95) ~[com-dancer-service-hollow.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.ExecutionException: com.netflix.hollow.api.error.IncompatibleSchemaException: No common schema exists for String: field value has unmatched types: bytes vs string
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
	at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
	at com.netflix.hollow.core.util.SimultaneousExecutor.awaitSuccessfulCompletion(SimultaneousExecutor.java:118) ~[hollow-2.16.6.jar:2.16.6]
	at com.netflix.hollow.core.write.HollowWriteStateEngine.restoreFrom(HollowWriteStateEngine.java:167) ~[hollow-2.16.6.jar:2.16.6]
	... 13 more
Caused by: com.netflix.hollow.api.error.IncompatibleSchemaException: No common schema exists for String: field value has unmatched types: bytes vs string
	at com.netflix.hollow.core.schema.HollowObjectSchema.findCommonSchema(HollowObjectSchema.java:173) ~[hollow-2.16.6.jar:2.16.6]
	at com.netflix.hollow.core.write.HollowTypeWriteState.restoreFrom(HollowTypeWriteState.java:307) ~[hollow-2.16.6.jar:2.16.6]
	at com.netflix.hollow.core.write.HollowWriteStateEngine$1.run(HollowWriteStateEngine.java:157) ~[hollow-2.16.6.jar:2.16.6]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
```
If I switch the JVM back to Java 8, the server starts without issue.  This is a major issue, as this server is box software and customers already have persisted state that we need to maintain.  However, we *would* like to move up to Java 11.  What should be done?

**Comment by brettwooldridge** [comment]
More information after some debugging...

One of the classes involved (code is Kotlin)...
```kotlin
@HollowPrimaryKey(fields = ["hostUuid"])
data class MonitorTarget(@HollowInline val hostUuid : String = "",
                         @HollowInline val ipAddress : String = "",
                         val network : String = "",
                         @HollowHashKey(fields = ["monitorId"]) val monitors : Set<Monitor?> = emptySet())
```
When breakpointing at ``HollowWriteStateEngine:134`` (version 2.16.6) and inspecting the various ``writeState`` values, I get...

JDK 8:
```
...
MonitorTarget @PrimaryKey(hostUuid) {
	string hostUuid;
	string ipAddress;
	String network;
	SetOfMonitor monitors;
}

String {
	string value;
}
```
JDK 11:
```
MonitorTarget @PrimaryKey(hostUuid) {
	string hostUuid;
	string ipAddress;
	String network;
	SetOfMonitor monitors;
}

String {
	bytes value;
}
```
The ``String`` schema seems to be ``string`` under JDK 8 and ``bytes`` under JDK 11...

**Comment by brettwooldridge** [comment]
Nevermind, this is addressed by #221, in v2.17.1 with support for Java compact strings.  Upgrading our dependency from v2.16.6 to v2.17.1 resolved the issue.  Maybe this will be useful for another user encountering this exception.

**Comment by PaulSandoz** [comment]
This does remind me though that Hollow should not be reflecting on the internals of JDK classes, i opened #347 to capture this.
